Content.Server/Resist/ResistLockerSystem.cs

```diff
@@ -0,0 +1,77 @@
+using Content.Shared.Movement;
+using Robust.Shared.GameObjects;
+using Content.Server.Storage.Components;
+using Content.Server.DoAfter;
+using Content.Shared.Popups;
+using Robust.Shared.Localization;
+
+namespace Content.Server.Resist.EntitySystems
+{
+    public class ResistLockerSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ResistLockerComponent, RelayMovementEntityEvent>(OnRelayMovement);
+        }
+
+
+        private void OnRelayMovement(EntityUid uid, ResistLockerComponent component, RelayMovementEntityEvent args)
+        {
+            if(EntityManager.TryGetComponent<EntityStorageComponent?>(uid, out var storageComponent))
+            {
+                if (!component.IsResisting)
+                {
+                    //Check IfWeldedShut or if Lock exists and is locked
+                    if (EntityManager.TryGetComponent<LockComponent?>(uid, out var lockComponent) && lockComponent.Locked || storageComponent.IsWeldedShut)
+                    {
+                            AttemptResist(args.Entity, uid, storageComponent, component);
+                    } 
+                }
+            }
+
+        }
+
+
+        private async void AttemptResist(EntityUid user, EntityUid target, EntityStorageComponent storageComponent, ResistLockerComponent resistLockerComponent)
+        {
```

Comment:
the component arguments to this function should be nullable, and resolved using

```csharp
if(!Resolve(target, ref storageComponent, ref resistLockerComponent))
   return;
```

Reply:
Hopefully i've handled this in a non-cursed way, i'm not too versed in C# but i've checked how we do it elsewhere in other systems